### ISSUE TITLE
Unable to create when condition for pin/unpin editor

### ISSUE THREAD

**Issue by Dimfred** [VSpaceCode/vscode-which-key#49]
Hey I am trying to create the pin / unpin logic, like with the toggle project tree example in the docs, but I can't make it work. Do you have any suggestions?

    {
        "key": "p",
        "name": "Pin / Unpin editor",
        "type": "conditional",
        "bindings": [
            {
                "key": "",
                "name": "default",
                "type": "command",
                "command": "workbench.action.pinEditor",
            },
            {
                "key": "when:activeEditorIsPinned",
                "name": "Unpin Editor",
                "type": "command",
                "command": "workbench.action.unpinEditor",
            },
        ]
    }

**Comment by stevenguh** [comment]
You will need to add the following to you shortcut (`keybindings.json`).
```
{
	"key": "p",
	"command": "whichkey.triggerKey",
	"args": {
		"key": "p",
		"when": "activeEditorIsPinned"
	},
	"when": "whichkeyVisible && activeEditorIsPinned"
}
```

We can also choose to bundle this as default so we don't have two separate key for pin and unpin.

**Comment by Dimfred** [comment]
Ah that is very sad. But thank you for the investigation, hope there will happen something soon, would be great to have that.

**Comment by stevenguh** [comment]
I am glad you are able to find a workaround.

I looked at the vscode api it doesn't seem like they provide anything to extension to know if the editor is pinned. To properly support this, we likely need to do a feature request to vscode for either

- Have programmatic access to the editor pinned state
- Add toggle command to toggle between pin and unpin

**Comment by Dimfred** [comment]
Yeah I think the second option is something they will do. If I have time I probably willl open an issue and request a unified command.
